lib/streamlit/elements/arrow.py

```diff
@@ -633,14 +633,23 @@ def dataframe(
     def table(self, data: Data = None) -> DeltaGenerator:
         """Display a static table.
 
-        This differs from ``st.dataframe`` in that the table in this case is
-        static: its entire contents are laid out directly on the page.
+        While ``st.dataframe`` is geared towards large datasets and interactive
+        data exploration, ``st.table`` is useful to display a small table without
+        scrolling – e.g. a confusion matrix or a leaderboard. For this reason, it also
+        supports Markdown formatting.
```

Comment:
@sfc-gh-dmatthews FYI I updated the description here to make it a bit easier to understand the differences.